test/Microsoft.ML.TimeSeries.Tests/TimeSeriesDirectApi.cs

```diff
@@ -571,6 +575,93 @@ public void AnomalyDetectionWithSrCnn(bool loadDataFromFile)
+                data.Add(new TimeSeriesDataDouble { Value = 10 });
+                for (int index = 0; index < 5; index++)
+                {
+                    data.Add(new TimeSeriesDataDouble { Value = 5 });
+                }
+
+                // Convert data to IDataView.
+                dataView = ml.Data.LoadFromEnumerable(data);
+            }
+
+            // Setup the detection arguments
+            string outputColumnName = nameof(SrCnnAnomalyDetection.Prediction);
+            string inputColumnName = nameof(TimeSeriesDataDouble.Value);
+
+            // Do batch anomaly detection
+            var outputDataView = ml.Data.BatchDetectAnomalyBySrCnn(dataView, outputColumnName, inputColumnName,
+                threshold: 0.35, batchSize: -1, sensitivity: 90.0, mode);
+
+            // Getting the data of the newly created column as an IEnumerable of
+            // SrCnnAnomalyDetection.
+            var predictionColumn = ml.Data.CreateEnumerable<SrCnnAnomalyDetection>(
+                outputDataView, reuseRowObject: false);
+
+            int k = 0;
+            foreach (var prediction in predictionColumn)
+            {
+                switch (mode)
+                {
+                    case SrCnnDetectMode.AnomalyOnly:
+                        Assert.Equal(3, prediction.Prediction.Length);
+                        if (k == 20)
+                            Assert.Equal(1, prediction.Prediction[0]);
+                        else
+                            Assert.Equal(0, prediction.Prediction[0]);
+                        break;
+                    case SrCnnDetectMode.AnomalyAndExpectedValue:
+                        Assert.Equal(4, prediction.Prediction.Length);
+                        if (k == 20)
+                        {
+                            Assert.Equal(1, prediction.Prediction[0]);
+                            Assert.Equal("5.00", prediction.Prediction[3].ToString("0.00"));
+                        }
+                        else
+                            Assert.Equal(0, prediction.Prediction[0]);
+                        break;
+                    case SrCnnDetectMode.AnomalyAndMargin:
+                        Assert.Equal(7, prediction.Prediction.Length);
+                        if (k == 20)
+                        {
+                            Assert.Equal(1, prediction.Prediction[0]);
+                            Assert.Equal("5.00", prediction.Prediction[3].ToString("0.00"));
+                            Assert.Equal("5.00", prediction.Prediction[4].ToString("0.00"));
+                            Assert.Equal("5.01", prediction.Prediction[5].ToString("0.00"));
+                            Assert.Equal("4.99", prediction.Prediction[6].ToString("0.00"));
+                        }
+                        else
+                            Assert.Equal(0, prediction.Prediction[0]);
+                        break;
+                }
+                k += 1;
+            }
+        }
+
```

Comment:
Can you file an issue and add a description to this PR?

_(pinned to a line to allow for discussions)_

Reply:
Done. Please see [https://github.com/dotnet/machinelearning/issues/5053](https://github.com/dotnet/machinelearning/issues/5053)

---
In reply to: [426822562](https://github.com/dotnet/machinelearning/pull/5135#discussion_r426822562) [](ancestors = 426822562)